metrics-monitoring/README.adoc

```diff
@@ -0,0 +1,225 @@
+:service-binding-url: service-discovery/README.adoc
+:access-mgmt-url: access-mgmt/README.adoc
+:access-mgmt-service-registry-url: access-mgmt-service-registry/README.adoc
+
+////
+END GENERATED ATTRIBUTES
+////
+
+[id="chap-monitoring-metrics"]
+= Monitoring metrics in {product-long}
+ifdef::context[:parent-context: {context}]
+:context: monitoring-metrics
+
+// Purpose statement for the assembly
+[role="_abstract"]
+As a developer or administrator, you can view metrics in {product} to visualize the performance and data usage for Kafka instances and topics that you have access to. You can view metrics directly in the {product} web console, or use the metrics API endpoint provided by {product} to import the data into your own metrics monitoring tool, such as Prometheus.
+
+//Additional line break to resolve mod docs generation error, not sure why. Leaving for now. (Stetson, 20 May 2021)
+
+[id="ref-supported-metrics_{context}"]
+== Supported metrics in {product}
+
+{product} supports the following metrics for Kafka instances and topics. In the {product} web console, a subset of these metrics is displayed in the *Dashboard* page of each Kafka instance.
+
+Broker metrics::
++
+--
+* `kafka_broker_quota_softlimitbytes`: Maximum amount of broker storage before producers are throttled
+
+* `kafka_broker_quota_totalstorageusedbytes`: Total storage used by the broker
+
+* `kafka_controller_kafkacontroller_global_partition_count`: Number of partitions across all topics in the cluster
+
+* `kafka_controller_kafkacontroller_offline_partitions_count`: Number of partitions without an active leader and that are therefore not writable or readable
+
+* `kafka_namespace:haproxy_server_bytes_in_total:rate5m`: *<@SME: Description?>*
+
+* `kafka_namespace:haproxy_server_bytes_out_total:rate5m`: *<@SME: Description?>*
+
+* `kafka_namespace:kafka_server_socket_server_metrics_connection_count:sum`: Number of client connections to the broker
+
+* `kafka_namespace:kafka_server_socket_server_metrics_connection_creation_rate:sum`: Number of attempted client connections per second for the broker
+
+* `kubelet_volume_stats_available_bytes`: Available disk space in the broker
+
+* `kubelet_volume_stats_used_bytes`: Used disk space in the broker
+--
+
+Topic metrics::
++
+--
+* `kafka_server_brokertopicmetrics_bytes_in_total`: Number of incoming bytes to topics in the instance
+
+* `kafka_server_brokertopicmetrics_bytes_out_total`: Number of outgoing bytes from topics in the instance
+
+* `kafka_server_brokertopicmetrics_messages_in_total`: Number of messages per second received by one or more topics in the instance
+
+* `kafka_topic:kafka_log_log_size:sum`: Log size for each partition in a topic *<@SME: This correct?>*
+
+* `kafka_topic:kafka_topic_partitions:count`: Number of partitions in a topic *<@SME: This correct?>*
+
+* `kafka_topic:kafka_topic_partitions:sum`: *<@SME: Description? I don't see the diff between this and "partitions:count".>*
```

Comment:
Interesting. Then what's the difference now between these two metrics, which sound the same to me? One is a broker metric, the other a topic one. Is the first one (broker) for all partitions across all instances, aka, across the cluster, and the second one (topic) just the partitions for all topics in one instance in that cluster? Not super clear because I've been hearing the Kafka world refer to a Kafka instance and Kafka cluster as the same thing, synonymous.

```
`kafka_controller_kafkacontroller_global_partition_count`: Number of partitions across all topics in the cluster

`kafka_topic:kafka_topic_partitions:sum`: Number of partitions across all topics in the instance
```

Reply:
I think the difference will be in the labels (which is what prompted me to ask where adding them would make things clearer).

Reply:
I'm not too familiar with the `kafka_controller_kafkacontroller_global_partition_count` metric but from looking at its results, it's grouped per broker whereas the `kafka_controller_kafkacontroller_global_partition_count` is partitions across the instance.

Reply:
I see. I've changed as follows. This work?

```
`kafka_controller_kafkacontroller_global_partition_count`: Number of partitions per broker in the instance

`kafka_topic:kafka_topic_partitions:sum`: Number of partitions across all topics in the instance
```

Reply:
yep that makes sense :+1: